hiscore: correctly look up ironmen players from chat message or self lookup

Hydrox6 opened the pull request:
Currently, when using the shortcut to lookup yourself on the hiscores (double clicking the magnifying glass), the normal account tab is used no matter what was set before.

Earlier, a conversation was had on Discord about getting this to also work off of chat message, since the information is available. The PR now covers this.

Adam- commented:
I think this has already been fixed for looking up yourself, since `resetEndpoints()` calls `selectWorldEndpoint()` which identifies the world type. We could still do the per-account type lookup though but this is conflicting atm.

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePanel.java:
```diff
@@ -138,8 +138,8 @@
-	public HiscorePanel(@Nullable Client client,
-		HiscoreConfig config, NameAutocompleter nameAutocompleter, OkHttpClient okHttpClient)
+	public HiscorePanel(@Nullable Client client, HiscorePlugin plugin, HiscoreConfig config,
+						NameAutocompleter nameAutocompleter, OkHttpClient okHttpClient)
```
```suggestion
		NameAutocompleter nameAutocompleter, OkHttpClient okHttpClient)
```

Nightfirecat commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java:
```diff
@@ -217,4 +231,25 @@ private void lookupPlayer(String playerName)
+
+	private HiscoreEndpoint getLocalHiscoreEndpointType()
+	{
+		EnumSet<WorldType> worldType = client.getWorldType();
+		if (worldType.contains(WorldType.LEAGUE))
+		{
+			return HiscoreEndpoint.LEAGUE;
+		}
```
This part of the logic could be replaced by `HiscorePanel#selectWorldEndpoint()`, e.g.:

```java
private HiscoreEndpoint getLocalHiscoreEndpoint()
{
	final HiscoreEndpoint profile = HiscorePanel.selectWorldEndpoint();
	if (profile != HiscoreEndpoint.NORMAL)
	{
		return profile;
	}

	switch (client.getAccountType())
	// ...
}
```

Hydrox6 commented:
I've moved some stuff around. 
* `selectWorldEndpoint` is now in the Plugin, renamed to `getWorldEndpoint`
* `getLocalHiscoreEndpointType` now has a null check, and uses `getWorldEndpoint`

Hydrox6 commented:
Earlier, a conversation was had on Discord about getting this to also work off of chat message, since the information is available. The PR now covers this.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java:
```diff
@@ -203,13 +207,22 @@ public void onMenuOptionClicked(MenuOptionClicked event)
 			{
+				// Prefer the world-based endpoint if the world is special
+				endpoint = getWorldEndpoint();
+				if (endpoint == HiscoreEndpoint.NORMAL)
```
Players can dm you from other worlds, so it probably would be best to just not check this. Or maybe check the flags of the world they are in, but those aren't always available.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java:
```diff
@@ -241,14 +254,19 @@ private void insertMenuEntry(MenuEntry newEntry, MenuEntry[] entries)
 	}
 
 	private void lookupPlayer(String playerName)
```
This is only use one other place, might aswell remove it and use the one which takes the endpoint.

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java:
```diff
@@ -296,4 +314,32 @@ private HiscoreEndpoint getLocalHiscoreEndpointType()
+		{
+			return HiscoreEndpoint.HARDCORE_IRONMAN;
+		}
+		return null;
```
I see no reason why this should ever be able to default to anything other than `NORMAL`

Adam- commented on runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java:
```diff
@@ -296,4 +314,32 @@ private HiscoreEndpoint getLocalHiscoreEndpointType()
+	private HiscoreEndpoint getHiscoreEndpointFromName(String rawName)
+	{
+		// Remove <col> tags, as they come before the <img> tags we need.
+		Matcher macther = COL_STRIPPER.matcher(rawName);
```
Instead of this, you could just contains() the name with IconID.toString()
